doc/doxygen/src/kconfig/kconfig.md

```diff
@@ -0,0 +1,366 @@
+These entries should depend on the module being used (see
+[Appendix A](#kconfig-appendix-a) to see how to check if a module is being
+used).
+
+The module configuration then can be enabled either via the menuconfig
+interface:
+
+![menuconfig-example](kconfig_menuconfig.png)
+
+or by means of a '.config' file:
+
+```Make
+CONFIG_KCONFIG_MODULE_GCOAP=y
+```
+
+---
+# Appendixes                                              {#kconfig-appendixes}
+
+## Appendix A: Check if a module or package is used       {#kconfig-appendix-a}
+In order to show only the relevant configuration parameters to the user with
+respect to a given application and board selection, Kconfig needs knowledge
+about all modules and packaged to be used for a compilation. Currently
+dependency handling among modules is performed by the build system (via
+`Makefile.dep` files). The interface defined to declared the used modules and
+packages is the `$ (GENERATED_DIR)/Kconfig.dep` file.
+
+`Kconfig.dep` is a Kconfig file that will define symbols of the form:
+```Kconfig
+config MODULE_SOCK_UTIL
+    bool
+    default y
+```
+There will be a symbol for every used module (i.e. every module in
+`USEMODULE` make variable) and package. The names in the symbols will be
+uppercase and separated by `_`. Based on these symbols configurability is
+decided. Modules and packages symbols will have `MODULE_` and `PKG_` prefixes
+respectively.
+
+The following is an example of how to use these symbols in Kconfig files to
+enable/disable a configuration menu:
+
+```Kconfig
+menuconfig KCONFIG_MODULE_SOCK_UTIL
+    bool "Configure Sock Utilities"
+    depends on MODULE_SOCK_UTIL
+    help
+        "Configure Sock Utilities using Kconfig."
+```
+
+Then, every configuration option for the previous module would be modeled like:
+
+```Kconfig
+if KCONFIG_MODULE_SOCK_UTIL
+
+config SOCK_UTIL_SCHEME_MAXLEN
+    int "Maximum length of the scheme part for sock_urlsplit"
+    default 16
+
+endif # KCONFIG_MODULE_SOCK_UTIL
+```
+
+## Appendix B: Difference between 'Kconfig' and '.config' files {#kconfig-appendix-b}
+Kconfig files describe a configuration database, which is a collection of
+configuration options organized in a tree structure. Configuration options have
+their own dependencies, which are used to determine its visibility.
+
+Kconfig files are written in
+[Kconfig language](https://www.kernel.org/doc/Documentation/kbuild/kconfig-language.txt)
+defined in the Linux kernel. Configuration options have attributes such as
+types, prompts and default values:
+
+```Kconfig
+menu "Buffer Sizes"
+
+config GCOAP_PDU_BUF_SIZE
+    int "Request or response buffer size"
+    default 128
+
+endmenu
+```
+
+On the other hand configuration files contain assignment of values to
+configuration options and use Makefile syntax:
+
+```makefile
+# enable Kconfig configuration for gcoap
+CONFIG_KCONFIG_MODULE_GCOAP=y
+# set the value
+CONFIG_GCOAP_PDU_BUF_SIZE=12345
+```
```

Comment:
You show two different types of files here (Kconf and Makefile) but in the compiled doc, this is not so obvious. Can you differentiate a bit better? By adding a caption or headings in separate lines, e.g *Kconf file* and *Configuration Makefile*